### ISSUE TITLE
Fail to build Python 3.5 with VS2015 and BUILD_EXTENSIONS_AS_BUILTIN 

### ISSUE THREAD

**Issue by Neurrone** [python-cmake-buildsystem/python-cmake-buildsystem#155]
Hi,

I'm trying to statically link extensions into libpython, so this project looks like it would fit what I need to do nicely. Just to test, I used the following cmake command, but then encountered the following visual studio errors:
```
cmake -G "Visual Studio 14" -DPYTHON_VERSION=3.5.2 -DCMAKE_BUILD_TYPE=Release -DBUILD_EXTENSIONS_AS_BUILTIN=ON ..
```
```
2>  _freeze_importlib.c
2>d:\source\python-cmake-buildsystem\python-3.5.2\modules_decimal\libmpdec\mpdecimal.h(231): fatal error C1189: #error:  "define CONFIG_64 or CONFIG_32"
```
And further down,
```
11>------ Skipped Build: Project: freeze_modules, Configuration: Release Win32 ------
11>Project not selected to build for this solution configuration 
10>  Building Custom Rule D:/source/python-cmake-buildsystem/cmake/libpython/CMakeLists.txt
...
10>  'Release_freeze_importlib.exe' is not recognized as an internal or external command,
10>  operable program or batch file.
10>C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppCommon.targets(171,5): error MSB6006: "cmd.exe" exited with code 9009.
```

**Comment by jcfr** [comment]
Thanks for the report, I will have a look later this week and post an update here.

**Comment by dbrnz** [comment]
Hmm, I suspect PR #152 would fix this.

**Comment by jcfr** [comment]
@dzenanz This issue and related PR should be relevant for your use case.

**Comment by dzenanz** [comment]
Thank you!

I will take a look when I get around to it.

Dženan Zukić, PhD, Senior R&D Engineer, Kitware (Carrboro, N.C.)

**Comment by droidmonkey** [comment]
PR #152 does not fix this issue. The problem is in the way _freeze_importlib works by assigning preprocessor defines to the source files. Unfortunately, mpdecimal.h is included in a slew of modules that have nothing to do with the _decimal module where the CONFIG_32/64 flag is defined in this cmake process.

The real fix is to add the flags resolved in ```${libmpdec_config_${libmpdec_config}}``` to the _freeze_importlib project.

**Comment by jcfr** [comment]
xref #161

**Comment by Neurrone** [comment]
@droidmonkey: I defined the following for that project:
CONFIG_32;PPRO;_CRT_SECURE_NO_WARNINGS;MASM;

In addition, I still had to #define CONFIG_32 in mpdecimal.h or it would claim that its undefined. Now the next error is:
Error	C2491	'_stricmp': definition of dllimport function not allowed	_freeze_importlib	D:\source\python-cmake-buildsystem\Python-3.5.2\Python\dynload_win.c	47

**Comment by jingtianming1** [comment]
I still have this bug in install spacy，this is ‘’mpdecimal.h(227): fatal error C1189: #error:  "define CONFIG_64 or CONFIG_32",if you can deal with it ,please help e

**Comment by jcfr** [comment]
Made progress on that issue.

Using this branch: https://github.com/python-cmake-buildsystem/python-cmake-buildsystem/compare/155-support-build-extensions-as-builtin-vs2015

and this set of patches:  https://github.com/python/cpython/compare/master...python-cmake-buildsystem:3.5.3-patches

bring the error count to 6.

```
            Creating library D:/T/python-cmake-buildsystem-build/CMakeBuild/libpython/Release/_freeze_importlib.lib and object D:/T/python-cmake-buildsystem-build/CMakeBuild/libpython/Release/_freeze_importlib.exp

     3>config.obj : error LNK2001: unresolved external symbol _PyInit_overlapped [D:\T\python-cmake-buildsystem-build\CMakeBuild\libpython\_freeze_importlib.vcxproj]

     3>_decimal.obj : error LNK2019: unresolved external symbol _mpd_del referenced in function _PyDec_AsTuple [D:\T\python-cmake-buildsystem-build\CMakeBuild\libpython\_freeze_importlib.vcxproj]

     3>io.obj : error LNK2001: unresolved external symbol _mpd_del [D:\T\python-cmake-buildsystem-build\CMakeBuild\libpython\_freeze_importlib.vcxproj]

     3>basearith.obj : error LNK2019: unresolved external symbol _mpd_uint_zero referenced in function __mpd_baseshiftl [D:\T\python-cmake-buildsystem-build\CMakeBuild\libpython\_freeze_importlib.vcxproj]

     3>io.obj : error LNK2019: unresolved external symbol _mpd_qresize referenced in function _mpd_qset_string [D:\T\python-cmake-buildsystem-build\CMakeBuild\libpython\_freeze_importlib.vcxproj]

     3>D:\T\python-cmake-buildsystem-build\CMakeBuild\libpython\Release\_freeze_importlib.exe : fatal error LNK1120: 4 unresolved externals [D:\T\python-cmake-buildsystem-build\CMakeBuild\libpython\_freeze_importlib.vcxproj]
```

**Comment by jcfr** [comment]
This Pr should address the issue: https://github.com/python-cmake-buildsystem/python-cmake-buildsystem/pull/197

After some more testing, it will be integrated.

Cc: @Neurrone @dbrnz @droidmonkey

**Comment by jcfr** [comment]
Closing. Fixed in https://github.com/python-cmake-buildsystem/python-cmake-buildsystem/commit/f55f8db60a81e9b7bd7138e5c87d59fcd779e287

**Comment by o01eg** [comment]
3.7, 3.8, and 3.9 are still affected by this issue.

**Comment by archibate** [comment]
> 3.7, 3.8, and 3.9 are still affected by this issue.

Hello everyone, I have fixed these issue for 3.9 in my fork: https://github.com/zenustech/python-cmake-buildsystem
